contracts/src/v0.8/test/HeartbeatRequester.t.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import 'forge-std/Test.sol';
+import {HeartbeatRequester} from "../HeartbeatRequester.sol";
+import '../mocks/MockAggregator.sol';
+import '../mocks/MockAggregatorProxy.sol';
+import '../mocks/MockOffchainAggregator.sol';
+
+    // from contracts directory,
+    // forge test --match-path src/v0.8/test/HeartbeatRequester.t.sol
+
+contract HeartbeatRequesterTest is Test {
+    HeartbeatRequester public heartbeatRequester;
+    MockAggregator public aggregator;
+    IAggregatorProxy public aggregatorProxy;
+
+    address internal constant USER = address(2);
+
+    event HeartbeatPermitted(address indexed permittedCaller, address newProxy, address oldProxy);
+    event HeartbeatRemoved(address indexed permittedCaller, address removedProxy);
+    error HeartbeatNotPermitted();
+
+    function setUp() public {
+        heartbeatRequester = new HeartbeatRequester();
+        aggregator = new MockAggregator();
+        aggregatorProxy = IAggregatorProxy(new MockAggregatorProxy(address(aggregator)));
+    }
+
+    function test_permitHeartbeat() public {
+        // Allow address(this) to request heartbeats from aggregatorProxy
+        vm.expectEmit(true, true, true, true);
+        emit HeartbeatPermitted(address(this), address(aggregatorProxy), address(0));
```

Comment:
instead of defining this event here again, can we re-use events in the contract? it seems it can.

https://book.getfoundry.sh/cheatcodes/expect-emit

Reply:
I think we need to define events in the test contract

<img width="745" alt="Screenshot 2023-06-16 at 1 53 28 PM" src="https://github.com/smartcontractkit/chainlink/assets/82334955/50d486fd-e435-46e3-9abc-dc2e48182438">


[https://ethereum.stackexchange.com/questions/141242/to-run-vm-expectemit-in-my-foundry-test-do-i-need-to-put-the-events-in-the-te](url)

Reply:
when doing
`emit SolidityFile.EventName(...)` it gives me an error, not sure why it works above. But either way I think we need to define it in the Test file

Reply:
got it